utils.pm

```diff
@@ -286,8 +286,14 @@ sub reboot
 	my ($self, $reason) = @_;
 
 	print("$reason, attempting to reboot...\n");
-	backend::reboot($self);
-	setup_ltp_run($self);
+	eval {
+		backend::reboot($self);
+		setup_ltp_run($self);
+	};
+	return 1 unless($@);
```

Comment:
I do not get why is the backend::reboot() and setup_ltp_run() executed in eval. Are we trying to avoid exiting on die() somewhere in the backend reboot handler?

Reply:
Yes, it is about line 365 in utils.pm to not die here, and this function actually return the results so they are stored in the result file, even if we run only the half.

Reply:
Hmm, there are couple of die statements in the ssh_start, I do not see them anywhere else. I guess that from a long term perspective we should probably just change these to log failure and return with error so that the error is propagated correctly.